ansible_runner/runner_config.py

```diff
@@ -314,13 +316,17 @@ def prepare_env(self):
         self.env = os.environ.copy()
         if self.envvars and isinstance(self.envvars, dict):
             if six.PY2:
-                self.env.update({k.decode('utf-8'):v.decode('utf-8') for k, v in self.envvars.items()})
+                envvars_update = {k.decode('utf-8'):v.decode('utf-8') for k, v in self.envvars.items()}
             else:
-                self.env.update({k:v for k, v in self.envvars.items()})
+                envvars_update = self.envvars
+            self.env_var_whitelist.extend(list(envvars_update.keys()))
+            self.env.update(envvars_update)
         try:
             envvars = self.loader.load_file('env/envvars', Mapping)
             if envvars:
-                self.env.update({six.text_type(k):six.text_type(v) for k, v in envvars.items()})
+                envvars_update = {six.text_type(k):six.text_type(v) for k, v in envvars.items()}
```

Comment:
Given that we are moving `devel` to py3 exclusively... does it make sense to carry `six` and use these features? Or is that still necessary?

Reply:
Should we try to pull `six` entirely?

Reply:
Yeah, no reluctance to removing six, but we still need to force type here because of bugs we've hit in the past with pexpect (for say, integer types). If it's a simple swap of `six.text_type` -> `str` then that's not a problem.